app/assets/stylesheets/govuk_publishing_components/components/_input.scss

```diff
@@ -1,10 +1,16 @@
-.gem-c-input {
+.gem-c-input,
+
+// Explicitly set the input type so that we have higher specificity than
+// https://github.com/alphagov/static/blob/79d29b6a4221874ead27e67d007704a6be166a57/app/assets/stylesheets/helpers/_buttons.scss#L91-L122
+// This can be removed once everthing is using `core_layout`.
+// scss-lint:disable QualifyingElement
+input[type=text].gem-c-input {
```

Comment:
Out of curiosity would doing `.gem-c-input.gem-c-input` by high enough specificity?